File: src/Atlasd/Battlenet/Protocols/Game/MessageContext.cs
using Atlasd.Battlenet.Sockets;
using System.Collections.Generic;

namespace Atlasd.Battlenet.Protocols.Game
{
    class MessageContext
    {
        public Dictionary<string, object> Arguments { get; protected set; }
        public TcpClient Client { get; protected set; }
        public MessageDirection Direction { get; protected set; }

        public MessageContext(TcpClient client, MessageDirection direction, Dictionary<string, object> arguments = null)
        {
            Arguments = arguments;
            Client = client;
            Direction = direction;
        }
    }
}

File: src/AtlasV/Battlenet/Protocols/Game/Other/MessageContext.vb
Imports System.Collections.Generic

Namespace AtlasV.Battlenet.Protocols.Game
    Class MessageContext
        Public Property Arguments As Dictionary(Of String, Object)
        Public Property Client As ClientState
        Public Property Direction As MessageDirection

        Public Sub New(ByVal varClient As ClientState,
                       ByVal varDirection As MessageDirection,
                       ByVal Optional varArguments As Dictionary(Of String, Object) = Nothing)
            Client = varClient
            Direction = varDirection
            Arguments = varArguments
        End Sub
    End Class
End Namespace